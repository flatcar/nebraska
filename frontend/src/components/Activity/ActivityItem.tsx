import { Box } from '@mui/material';
import Grid from '@mui/material/Grid';
import Link from '@mui/material/Link';
import ListItem from '@mui/material/ListItem';
import Typography from '@mui/material/Typography';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link as RouterLink } from 'react-router-dom';

import { Activity } from '../../api/apiDataTypes';
import { toLocaleString } from '../../i18n/dateTime';
import { activityStore } from '../../stores/Stores';
import ActivityItemIcon from './ActivityItemIcon';

export interface ActivityItemProps {
  entry: Activity;
}

export default function ActivityItem(props: ActivityItemProps) {
  const entryClass = activityStore().makeActivityEntryClass(props.entry.class, props.entry);
  const entrySeverity = activityStore().makeActivityEntrySeverity(props.entry.severity);

  return (
    <ActivityItemPure
      createdTs={props.entry.created_ts}
      appId={props.entry.app_id}
      groupId={props.entry.group_id}
      classType={entryClass.type}
      groupName={entryClass.groupName}
      appName={entryClass.appName}
      description={entryClass.description}
      severityName={entrySeverity.className}
    />
  );
}

export interface ActivityItemPureProps {
  appId: string;
  appName: string;
  classType: string;
  createdTs: string;
  description: string | React.ReactElement<any, string | React.JSXElementConstructor<any>>;
  groupId: string;
  groupName: string | null;
  severityName: string;
}

export function ActivityItemPure(props: ActivityItemPureProps) {
  const { t } = useTranslation();

  const time = toLocaleString(props.createdTs, undefined, {
    hour: '2-digit',
    minute: '2-digit',
  });
  let subtitle = '';
  let name: React.ReactNode = '';

  if (props.classType !== 'activityChannelPackageUpdated') {
    const groupPath = `apps/${props.appId}/groups/${props.groupId}`;
    subtitle = t('activity|group');
    name = (
      <Link component={RouterLink} to={groupPath} sx={{ color: '#1b5c91' }} underline="hover">
        {props.groupName}
      </Link>
    );
  }

  return (
    <ListItem
      alignItems="flex-start"
      disableGutters
      sx={{
        paddingTop: '15px',
        paddingLeft: '15px',
      }}
    >
      <Grid container alignItems="center" justifyContent="space-between">
        <Grid size={10}>
          <Box display="flex" alignItems="center" justifyContent="flex-start">
            <Box mr={1}>
              <ActivityItemIcon severityName={props.severityName} />
            </Box>
            <Box>
              <Typography
                sx={{
                  fontWeight: 'bold',
                  fontSize: '1.1rem',
                  color: '#474747',
                }}
              >
                {props.appName}
              </Typography>
            </Box>
          </Box>
        </Grid>
        <Grid size={2}>
          <Typography
            color="textSecondary"
            sx={{
              fontSize: '.7rem',
            }}
          >
            {time}
          </Typography>
        </Grid>
        {subtitle && (
          <Grid container spacing={2} size={12}>
            <Grid>
              <Typography color="textSecondary" display="inline">
                {subtitle}
              </Typography>
            </Grid>
            <Grid>{name}</Grid>
          </Grid>
        )}
        <Grid>{props.description}</Grid>
      </Grid>
    </ListItem>
  );
}
