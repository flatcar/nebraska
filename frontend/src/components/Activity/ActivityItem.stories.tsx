import { Meta, StoryFn } from '@storybook/react';
import { MemoryRouter } from 'react-router-dom';

import { ActivityItemPure, ActivityItemPureProps } from './ActivityItem';

export default {
  title: 'activity/ActivityItem',
} as Meta;

const Template: StoryFn<ActivityItemPureProps> = args => (
  <MemoryRouter>
    <ActivityItemPure {...args} />
  </MemoryRouter>
);

export const Warning = {
  render: Template,

  args: {
    createdTs: '2020-05-13T20:26:03.837688+05:30',
    appId: 'XXXX-XXX',
    groupId: 'YYYY-YYYY',
    classType: 'someType',
    groupName: 'some group',
    appName: 'some app',
    description: 'A description',
    severityName: 'warning',
  },
};

export const ActivityChannelPackageUpdated = {
  render: Template,

  args: {
    createdTs: '2020-05-13T20:26:03.837688+05:30',
    appId: 'XXXX-XXX',
    groupId: 'YYYY-YYYY',
    classType: 'activityChannelPackageUpdated',
    groupName: 'some group',
    appName: 'some app',
    description: 'A description',
    severityName: 'info',
  },
};
