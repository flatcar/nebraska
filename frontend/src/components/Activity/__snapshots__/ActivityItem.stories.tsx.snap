// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activity/ActivityItem ActivityChannelPackageUpdated smoke-test 1`] = `
<li class="MuiListItem-root MuiListItem-padding MuiListItem-alignItemsFlexStart css-kd2s28-MuiListItem-root">
  <div class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-tqvfrr-MuiGrid-root">
    <div class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-10 css-r13beu-MuiGrid-root">
      <div class="MuiBox-root css-geek62">
        <div class="MuiBox-root css-12z0wuy">
          <svg xmlns="http://www.w3.org/2000/svg"
               xmlns:xlink="http://www.w3.org/1999/xlink"
               aria-hidden="true"
               role="img"
               width="30px"
               height="30px"
               viewbox="0 0 24 24"
               style="color: rgb(0, 211, 255);"
          >
            <path fill="currentColor"
                  d="M11 15h2v2h-2v-2m0-8h2v6h-2V7m1-5C6.47 2 2 6.5 2 12a10 10 0 0 0 10 10a10 10 0 0 0 10-10A10 10 0 0 0 12 2m0 18a8 8 0 0 1-8-8a8 8 0 0 1 8-8a8 8 0 0 1 8 8a8 8 0 0 1-8 8Z"
            >
            </path>
          </svg>
        </div>
        <div class="MuiBox-root css-0">
          <p class="MuiTypography-root MuiTypography-body1 css-10o1zsr-MuiTypography-root">
            some app
          </p>
        </div>
      </div>
    </div>
    <div class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-2 css-15bfe7d-MuiGrid-root">
      <p class="MuiTypography-root MuiTypography-body1 css-1nwggdl-MuiTypography-root">
        02:56 PM
      </p>
    </div>
    <div class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root">
      A description
    </div>
  </div>
</li>
`;

exports[`activity/ActivityItem Warning smoke-test 1`] = `
<li class="MuiListItem-root MuiListItem-padding MuiListItem-alignItemsFlexStart css-kd2s28-MuiListItem-root">
  <div class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row css-tqvfrr-MuiGrid-root">
    <div class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-10 css-r13beu-MuiGrid-root">
      <div class="MuiBox-root css-geek62">
        <div class="MuiBox-root css-12z0wuy">
          <svg xmlns="http://www.w3.org/2000/svg"
               xmlns:xlink="http://www.w3.org/1999/xlink"
               aria-hidden="true"
               role="img"
               width="30px"
               height="30px"
               viewbox="0 0 24 24"
               style="color: rgb(255, 85, 0);"
          >
            <path fill="currentColor"
                  d="M12 2L1 21h22M12 6l7.53 13H4.47M11 10v4h2v-4m-2 6v2h2v-2"
            >
            </path>
          </svg>
        </div>
        <div class="MuiBox-root css-0">
          <p class="MuiTypography-root MuiTypography-body1 css-10o1zsr-MuiTypography-root">
            some app
          </p>
        </div>
      </div>
    </div>
    <div class="MuiGrid-root MuiGrid-direction-xs-row MuiGrid-grid-xs-2 css-15bfe7d-MuiGrid-root">
      <p class="MuiTypography-root MuiTypography-body1 css-1nwggdl-MuiTypography-root">
        02:56 PM
      </p>
    </div>
    <div class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-row MuiGrid-grid-xs-12 MuiGrid-spacing-xs-2 css-1314ia1-MuiGrid-root">
      <div class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root">
        <p class="MuiTypography-root MuiTypography-body1 css-17imrw3-MuiTypography-root">
          GROUP
        </p>
      </div>
      <div class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root">
        <a class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-1mxnekj-MuiTypography-root-MuiLink-root"
           href="/apps/XXXX-XXX/groups/YYYY-YYYY"
        >
          some group
        </a>
      </div>
    </div>
    <div class="MuiGrid-root MuiGrid-direction-xs-row css-17fpwt7-MuiGrid-root">
      A description
    </div>
  </div>
</li>
`;
